consensus/src/ledger/mod.rs

```diff
@@ -77,6 +78,8 @@ pub trait Ledger: Send + Sync {
 
     /// checks if a ledgers state is consistent
     fn validate_ledger(&self) -> bool;
+
+    fn requires_full_rebuilds(&self, new_commitments_len: usize, new_serial_numbers_len: usize) -> bool;
```

Comment:
can we rename this to something like `require_async_task` to be less context-specific.